programs/server/Server.cpp

```diff
@@ -562,6 +560,35 @@ int Server::main(const std::vector<std::string> & /*args*/)
 
     StatusFile status{path + "status", StatusFile::write_full_info};
 
+
+    /// Write a uuid file containing a unique uuid if the file doesn't already exist during server start.
+    {
+        fs::path server_uuid_file(path + "uuid");
+
+        if (!fs::exists(server_uuid_file))
+        {
+            try
+            {
+                /// Note: Poco::UUIDGenerator().createRandom() uses /dev/random and can be expensive. But since
+                /// it's only going to be generated once (i.e if the uuid file doesn't exist), it's probably fine.
+                auto uuid_str = Poco::UUIDGenerator().createRandom().toString();
+                WriteBufferFromFile out(server_uuid_file.string());
+                out.write(uuid_str.data(), uuid_str.size());
+                out.sync();
+                out.finalize();
+            }
+            catch (...)
+            {
+                throw Poco::Exception("Caught Exception while writing to write UUID file {}.\n", server_uuid_file.string());
```

Comment:
It is losing the initial context of exception.

Reply:
Fixed - added exception message to the Exception thrown now.

Reply:
Preserving and throwing exception.